File: implementation/scipamato/public/web/src/main/java/ch/difty/scipamato/publ/web/pym/PymScripts.java
package ch.difty.scipamato.publ.web.pym;

import org.jetbrains.annotations.NotNull;

/**
 * Enum defining the JavaScript snippets used in the context of
 * providing responsive iframes with pym.js.
 */
public enum PymScripts {
    INSTANTIATE("pymChild", "var pymChild = new pym.Child({ id: 'scipamato-public' });"),
    RESIZE("pymResize", "pymChild.sendHeight();");

    public final String id;
    public final String script;

    PymScripts(@NotNull final String id, @NotNull final String script) {
        this.id = id;
        this.script = script;
    }
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/pym/PymScripts.kt
package ch.difty.scipamato.publ.web.pym

/**
 * Enum defining the JavaScript snippets used in the context of providing responsive iframes with pym.js.
 */
enum class PymScripts(val id: String, val script: String) {
    INSTANTIATE("pymChild", "var pymChild = new pym.Child({ id: 'scipamato-public' });"),
    RESIZE("pymResize", "pymChild.sendHeight();");
}
